docs/source/developer_api.rst

```diff
@@ -13,32 +13,121 @@ metadata
-	:members:
+    :members:
+
+MD potentials
+=============
+
+.. automodule:: brer.md
+
+As of 2022, we are still using the old gmxapi 0.0.7 style
+`WorkElement <https://gmxapi.readthedocs.io/en/release-0_0_7/reference.html#gmx.workflow.WorkElement>`__
+to connect pluggable extension code to the GROMACS simulator.
+The WorkElement is constructed by the `brer.plugin_configs` module by encoding
+an entry point function and its input parameters.
+
+Entry points
+------------
+
+The following creation functions are used to support creation of gmxapi
+pluggable GROMACS restraint potentials.
+The entry point function consumes a
+:py:class:`gmxapi.simulation.workflow.WorkElement`
+to produce an opaque object that can finish setting up the interactions with the
+gmxapi simulator lifetime management.
+
+In addition to the parameters described below, all restraint potentials require
+a :term:`sites` parameter.
+
+.. glossary::
+
+    sites
+        (*list[int]*) Describe the pair distance relevant to an instance of a BRER potential.
+        The list is a sequence of two or more sites, expressed as atom indices
+        in the molecular model. The first and last index define the origin and tip
+        of the displacement vector. In the case that this vector may be ambiguous
+        with respect to periodic boundary conditions (if the sites are likely to
+        ever be further apart on the molecule than half of the shortest simulation
+        box dimension), additional indices can be inserted between the first and
+        last element to define a sequence of vectors that can reliably sum to the
+        intended vector.
+
+.. todo:: Auto-extract the parameter structure docs once parameters are expressed more cleanly in the C++ code.
+
+.. py:function:: linear_restraint()
+
+    Configure the BRER potential used for the production phase.
+
+    :param float alpha: coupling parameter.
+    :param float target: site displacement distance to bias towards.
+    :param float sample_period: time between samples (simulator time units)
+    :param str logging_filename: output file
+
+.. todo:: Get parameters from makeLineareStopParams docstring.
+
+.. py:function:: linearstop_restraint()
+
+    Configure the BRER potential used during the convergence phase.
+
+    :param float alpha: coupling parameter.
+    :param float target: site displacement distance to bias towards.
+    :param float tolerance: convergence detection parameter
+    :param float sample_period: time between samples (simulator time units)
+    :param str logging_filename: output file
+
+.. autofunction:: brer_restraint()
+
+    Configure the BRER potential used for the training phase.
+
+    :param float A: training parameter.
+    :param float tau: training parameter.
+    :param float tolerance: convergence detection parameter
+    :param float target: site displacement distance to bias towards.
+    :param float sample_period: time between samples (simulator time units)
```

Comment:
TODO: Double-check this. I think I may have mistranscribed the parameters. looking at brerpotential.cpp, it looks like nSamples is input, not sample_period.

Reply:
I don't know why this is a floating point value instead of an integer, but I have double-checked the code. (Integers will, of course be converted (without rounding error for normal use cases).)
```suggestion
    :param float num_samples: number of samples to average at each update.
```